_sources/Unit1-Using-Objects-and-Methods/topic-1-7-APIs-and-libraries.rst

```diff
@@ -1,10 +1,202 @@
+
+.. mchoice:: built_in_package
+    :answer_a: java.util
+    :answer_b: java.io
+    :answer_c: java.lang
+    :answer_d: java.net
+    :correct: c
+    :feedback_a: Incorrect. java.util is not built into Java.
+    :feedback_b: Incorrect. java.io is not built into Java.
+    :feedback_c: Correct! java.lang is built into Java.
+    :feedback_d: Incorrect. java.net is not built into Java.
+
+    Which package is already built into Java and does not need to be imported?
+
+
+.. mchoice:: system_class
+    :answer_a: String
+    :answer_b: System
+    :answer_c: PrintStream
+    :answer_d: Scanner
+    :correct: b
+    :feedback_a: Incorrect. The String class does not have methods to interact with the operating system.
+    :feedback_b: Correct! The System class has methods to interact with the operating system.
+    :feedback_c: Incorrect. The PrintStream class does not have methods to interact with the operating system.
+    :feedback_d: Incorrect. The Scanner class does not have methods to interact with the operating system.
+
+    Which class in the java.lang package has methods to interact with the computer's operating system?
+
+.. mchoice:: example_method
+    :answer_a: System
+    :answer_b: String
+    :answer_c: java.lang
+    :answer_d: println()
+    :correct: d
+    :feedback_a: Incorrect. System is not a method.
+    :feedback_b: Incorrect. String is not a method.
+    :feedback_c: Incorrect. java.lang is not a method.
+    :feedback_d: Correct! println() is an example of a method.
+
+    Which of the following is an example of a method in Java?
+
+.. mchoice:: system_attributes
+    :answer_a: in
+    :answer_b: input
+    :answer_c: print
+    :answer_d: println
+    :correct: a
+    :feedback_d: Incorrect. println is a method, not an attribute.
+    :feedback_b: Incorrect. input is not an attribute in the System class.
+    :feedback_a: Correct! The System class has three fields: in, out, and err.
+    :feedback_c: Incorrect. print is not an attribute in the System class.
+
+    Which of the following is an attribute in the ``System`` class?
+
+.. mchoice:: purpose_apis
+    :answer_a: To write new code from scratch
+    :answer_b: To use code written by others
+    :answer_c: To create programming languages
+    :answer_d: To compile code
+    :correct: b
+    :feedback_a: Incorrect. APIs and libraries allow you to use code written by others.
+    :feedback_b: Correct! APIs and libraries are used to use code written by others.
+    :feedback_c: Incorrect. APIs and libraries are not for creating programming languages.
+    :feedback_d: Incorrect. APIs and libraries are not for compiling code.
+
+    What is the purpose of APIs and libraries in programming?
+
+.. mchoice:: attributes_class
+    :answer_a: Methods that perform tasks
+    :answer_b: Data related to the class stored in variables
+    :answer_c: Interfaces related to the class
```

Comment:
This curriculum does't cover interfaces right. Probably need to change this question.